src/core_functions/scalar/list/list_distance.cpp

```diff
@@ -0,0 +1,70 @@
+#include "duckdb/core_functions/scalar/list_functions.hpp"
+#include <cmath>
+
+namespace duckdb {
+
+template <class NUMERIC_TYPE>
+static void ListDistance(DataChunk &args, ExpressionState &, Vector &result) {
+	D_ASSERT(args.ColumnCount() == 2);
+
+	auto count = args.size();
+	auto &left = args.data[0];
+	auto &right = args.data[1];
+	auto left_count = ListVector::GetListSize(left);
+	auto right_count = ListVector::GetListSize(right);
+
+	UnifiedVectorFormat left_child_format;
+	ListVector::GetEntry(left).ToUnifiedFormat(left_count, left_child_format);
+	if (!left_child_format.validity.CheckAllValid(left_count)) {
```

Comment:
The `ToUnifiedFormat` here is somewhat misleading as the selection vector is ignored - neither in `CheckAllValid` nor in the main loop. As a result this code only works when the list child is a flat vector. I believe we enforce list children to be flat vectors now always anyway so it doesn't matter - but perhaps we can be explicit about this and use the `FlatVector` methods instead?

Reply:
Good point, I really just wanted some way to access the validity. I've swapped to the FlatVector accessors and added asserts for the child vector type, but AFAIK you're right about list children always being flat.